lib/util/device_status_ext.dart

```diff
@@ -11,11 +11,39 @@ extension ListDeviceStatusExtension
     final thisDevice = canvasClientServiceV2.clientDeviceInfo;
     for (final devicePair in this) {
       final status = devicePair.second;
-      if (status.connectedDevice.deviceId == thisDevice.deviceId) {
-        controllingDeviceStatus[devicePair.first.id] = status;
-        break;
+      if (status.connectedDevice?.deviceId == thisDevice.deviceId) {
+        controllingDeviceStatus[devicePair.first.deviceId] = status;
       }
     }
     return controllingDeviceStatus;
   }
 }
+
+extension DeviceStatusExtension on CheckDeviceStatusReply {
+  String? get playingArtworkKey {
+    if (artworks.isEmpty && exhibitionId == null) {
+      return null;
+    }
+    if (exhibitionId != null) {
+      return exhibitionId.toString();
+    }
+
+    final hashCode = artworks.playArtworksHashCode;
+    return hashCode.toString();
+  }
+}
+
+extension PlayArtworksExtension on List<PlayArtworkV2> {
+  int get playArtworksHashCode {
```

Comment:
Should it just be hashCode?

Reply:
this should return same when calculate hashcode from playlist